catkin_ws/src/planner/src/substates/utility/controller.py

```diff
@@ -18,6 +18,7 @@
 do_txtytz = [Bool(False),Bool(False),Bool(False),Bool(True),Bool(True),Bool(True)]
 do_tz = [Bool(False),Bool(False),Bool(False),Bool(False),Bool(False),Bool(True)]
 do_all = [Bool(True),Bool(True),Bool(True),Bool(True),Bool(True),Bool(True)]
+quaternion_dos = [Bool(True),Bool(True),Bool(True),Bool(True)]
 do_displace = Bool(True)
 do_not_displace = Bool(False)
 
```

Comment:
having all of these definitions for different combinations of true/false seems overly complicated, why not 5 global variables: do_x, do_y, do_z, do_rot, do_displace?

Reply:
or nevermind I see now they're used to avoid defining them manually, but there has to be a better way. With superimposer + quaternion (I'm assuming we only have those 2), 6 DOFs for superimposer and 4 DOFs for quaternion, that's 6^2=36 options for superimposer, 4^2=16 options for quaternion, minus some because you can't have do_x on both at the same time, but you get my point